e3/anod/context.py

```diff
@@ -438,7 +438,10 @@ def add_dep(spec_instance, dep, dep_instance):
                 # spec has no binary package to download.
                 raise SchedulingError(
                     "error in plan at {}: "
-                    "install should be replaced by build".format(plan_line))
+                    "Cannot install `{}' because this spec does not produce "
+                    "a package; check you are using the appropriate "
```

Comment:
this is misleading: it is not that the spec does not define a package, it is that the spec does the define a way of downloading a binary package.